src/ast/passes/codegen_llvm.cpp

```diff
@@ -264,17 +264,22 @@ void CodegenLLVM::visit(Builtin &builtin)
     }
 
     int arg_num = atoi(builtin.ident.substr(4).c_str());
-    Value *ctx = b_.CreatePointerCast(ctx_, b_.getInt64Ty()->getPointerTo());
-    Value *sp = b_.CreateLoad(
-        b_.getInt64Ty(),
-        b_.CreateGEP(b_.getInt64Ty(), ctx, b_.getInt64(sp_offset)),
-        "reg_sp");
-    dyn_cast<LoadInst>(sp)->setVolatile(true);
+    Value *sp = b_.CreateRegisterRead(ctx_, sp_offset, "reg_sp");
     AllocaInst *dst = b_.CreateAllocaBPF(builtin.type, builtin.ident);
-    Value *src = b_.CreateAdd(sp,
-                              b_.getInt64((arg_num + arch::arg_stack_offset()) *
-                                          sizeof(uintptr_t)));
-    b_.CreateProbeRead(ctx_, dst, 8, src, builtin.type.GetAS(), builtin.loc);
+
+    // Pointer width is used when calculating the SP offset and the number of
+    // bytes to read from stack for each argument (e.g. on 32-bit systems sargN
+    // 4 bytes). We pass a pointer SizedType to CreateProbeRead to make sure it
```

Comment:
The part in the parentheses needs rewording but also the formula is right there in the code so the example is redundant